python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py

```diff
@@ -444,53 +443,58 @@ def _process_sampled_df_csc(
         destinations, respectively.
     """
     # dropna
-    major_offsets = df.major_offsets.dropna().values
-    label_hop_offsets = df.label_hop_offsets.dropna().values
-    renumber_map_offsets = df.renumber_map_offsets.dropna().values
-    renumber_map = df.map.dropna().values
-    minors = df.minors.dropna().values
+    major_offsets = cast_to_tensor(df.major_offsets.dropna())
+    label_hop_offsets = cast_to_tensor(df.label_hop_offsets.dropna())
+    renumber_map_offsets = cast_to_tensor(df.renumber_map_offsets.dropna())
+    renumber_map = cast_to_tensor(df.map.dropna())
+    minors = cast_to_tensor(df.minors.dropna())
 
-    n_batches = renumber_map_offsets.size - 1
-    n_hops = int((label_hop_offsets.size - 1) / n_batches)
+    n_batches = len(renumber_map_offsets) - 1
+    n_hops = int((len(label_hop_offsets) - 1) / n_batches)
 
     # make global offsets local
-    major_offsets -= major_offsets[0]
-    label_hop_offsets -= label_hop_offsets[0]
-    renumber_map_offsets -= renumber_map_offsets[0]
+    # Have to make a clone as pytorch does not allow
+    # in-place operations on tensors
+    major_offsets -= major_offsets[0].clone()
+    label_hop_offsets -= label_hop_offsets[0].clone()
+    renumber_map_offsets -= renumber_map_offsets[0].clone()
 
     # get the sizes of each adjacency matrix (for MFGs)
     mfg_sizes = (label_hop_offsets[1:] - label_hop_offsets[:-1]).reshape(
         (n_batches, n_hops)
     )
     n_nodes = renumber_map_offsets[1:] - renumber_map_offsets[:-1]
-    mfg_sizes = cupy.hstack((mfg_sizes, n_nodes.reshape(n_batches, -1)))
+    mfg_sizes = torch.hstack((mfg_sizes, n_nodes.reshape(n_batches, -1)))
     if reverse_hop_id:
-        mfg_sizes = mfg_sizes[:, ::-1]
+        mfg_sizes = mfg_sizes.flip(1)
 
     tensors_dict = {}
     renumber_map_list = []
+    # Note: minors and major_offsets from BulkSampler are of type int32
+    # and int64 respectively. Since pylibcugraphops binding code doesn't
+    # support distinct node and edge index type, we simply casting both
+    # to int32 for now.
+    minors = minors.int()
+    major_offsets = major_offsets.int()
+    # Note: We transfer tensors to CPU here to avoid the overhead of
+    # transferring them in each iteration of the for loop below.
+    major_offsets_cpu = major_offsets.to("cpu").numpy()
+    label_hop_offsets_cpu = label_hop_offsets.to("cpu").numpy()
```

Comment:
@tingyu66 , This is the main optimization because transferring b/w CPU ->GPU 1 tensor at a time was slow.

Reply:
Thanks for resolving the `.item()` overhead. :+1: